python/cell_reader.py

```diff
@@ -195,6 +196,69 @@ def get_factors(metadata):
     return factors
 
 
+def get_cell_sets(clusters, lookup):
+    '''
+    >>> Cluster = namedtuple('Cluster', ['name', 'cell_ids'])
+    >>> clusters = {
+    ...   1: Cluster('pyramidal L4', ['104', '110', '111']),
+    ...   3: Cluster('vascular smooth muscle', ['1', '2', '3'])
+    ... }
+    >>> lookup = {
+    ...   'vascular smooth muscle': 'vasculature',
+    ...   'pyramidal L4': 'excitatory neurons'
+    ... }
+    >>> cell_sets = get_cell_sets(clusters, lookup)
+    >>> list(cell_sets.keys())
+    ['version', 'datatype', 'tree']
+    >>> cell_sets['datatype']
+    'cell'
+    >>> list(cell_sets['tree'][0].keys())
+    ['name', 'children']
+    >>> cell_sets['tree'][0]['name']
+    'Cell Type Annotations'
+    >>> sorted([ n['name'] for n in cell_sets['tree'][0]['children'] ])
+    ['excitatory neurons', 'vasculature']
+    '''
+
+    cluster_name_to_cell_ids = dict(
+        (c.name, c.cell_ids) for c in clusters.values()
+    )
+
+    hierarchy = dict([
+        (c, dict([
+            (sc, cluster_name_to_cell_ids[sc])
+            for sc, sc_c in lookup.items() if sc_c == c
+        ]))
+        for c in lookup.values()
+    ])
```

Comment:
Took be a bit to understand what was going on here. Would something like this work? 

Probably not that important, but having shared variable names between the creation and traversal of the hierarchy might helpful for future readers, and lookup is used as a the hash table. If failing loudly is desired, I'd replace `Dictionary.get` with `Dictionary[]`, since the former returns `None` if a key is missing.

```suggestion
    hierarchy = {cluster_name: {} for cluster_name in lookup.values()}
    for c in clusters.values():
        subcluster = {c.name: c.cell_ids}
        cluster_name = lookup.get(c.name)
        cluster_dict = hierarchy.get(cluster_name)
        cluster_dict.update(subcluster)
```

Reply:
Thanks for the suggestion! I have adopted this change and added comments to describe more clearly what the code is doing / what the variables contain in the latest commit https://github.com/hubmapconsortium/vitessce-data/pull/94/commits/28ea851898e1d58c5cd3725709de77f9b29cc915